src/audio_core/lle/lle.cpp

```diff
@@ -177,15 +173,26 @@ struct DspLle::Impl final {
         }
     }
 
-    void TeakraSliceEvent(u64 late) {
-        RunTeakraSlice();
-        u64 next = TeakraSlice * 2; // DSP runs at clock rate half of the CPU rate
-        if (next < late)
-            next = 0;
-        else
-            next -= late;
-        Core::System::GetInstance().CoreTiming().ScheduleEvent(next, teakra_slice_event, 0);
-    }
+    class TeakraSliceEvent : public Core::Event {
+        Impl& parent;
+
+    public:
+        TeakraSliceEvent(Impl& parent_) : parent(parent_) {}
+        const std::string& Name() const override {
+            return "DSP slice";
+        }
+        void Execute(Core::Timing& timing, u64 userdata, Ticks late) {
+            parent.RunTeakraSlice();
+            auto next = Ticks(TeakraSlice * 2); // DSP runs at clock rate half of the CPU rate
+            if (next < late)
+                next = Ticks(0);
+            else
+                next = next - late;
+            timing.ScheduleEvent(this, next);
+        }
+    };
+
+    std::unique_ptr<TeakraSliceEvent> teakra_slice_event{new TeakraSliceEvent(*this)};
```

Comment:
We generally use make_unique instead of new, even if it's right next to the type being constructed.

```suggestion
    std::unique_ptr<TeakraSliceEvent> teakra_slice_event = std::make_unique<TeakraSliceEvent>(*this);
```